kmip/core/factories/secrets.py

```diff
@@ -132,7 +103,52 @@ def _create_template(self, value):
             return Template(value)
 
     def _create_secret_data(self, value):
-        raise NotImplementedError()
+        if value:
```

Comment:
See my comments on the latest commit. Forgot we needed to swap the `if` blocks around.